### PR TITLE
cpychecker fixes & header cleanups

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
@luizirber @camillescott @ctb Ready for review & merge

**Comment by mr-c** [comment]
Please, can I get a review? @ctb

**Comment by mr-c** [comment]
Jenkins, retest this please

**Comment by mr-c** [comment]
Passes all tests, code coverage is lowered due to no way to test some of the error routes.

**Comment by ctb** [comment]
@luizirber can you take a look at this?

File: lib/read_parsers.cc [hunk 1]
```diff
@@ -5,14 +5,26 @@
...
+#include <fstream>
 
-#include <cstring>
 #include "khmer_exception.hh"
-#include <seqan/sequence.h>
-#include <seqan/seq_io.h>
-#include <seqan/stream.h>
-#include <pthread.h>
+#include "read_parsers.hh"
+#include "seqan/basic/allocator_interface.h"
```
Comment (luizirber) [inline lib/read_parsers.cc]: Do this need to be explicitly imported, even with the seqan includes on the previous block?

File: Makefile [hunk 2]
```diff
@@ -160,9 +160,9 @@ diff_pylint_report: pylint_report.txt
 # We need to get coverage to look at our scripts. Since they aren't in a
 # python module we can't tell nosetests to look for them (via an import
 # statement). So we run nose inside of coverage.
-.coverage: $(PYSOURCES) $(wildcard tests/*.py) khmer/_khmermodule.so
+.coverage: $(PYSOURCES) $(wildcard tests/*.py) khmer/_khmer*.so
 	coverage run --branch --source=scripts,khmer,oxli --omit=khmer/_version.py \
-		-m nose --with-xunit --attr=\!known_failing --processes=0
+		-m nose --with-xunit --attr=${TESTATTR} --processes=0
```
Comment (luizirber) [inline Makefile]: These changes on the Makefile will conflict with #1108

**Comment by luizirber** [comment]
Overall, LGTM. Are you planning any other changes?

**Comment by mr-c** [comment]
Nope, just wanting to get these two month old commits merged :-) I'll bring it up to date once #1108 is in.

**Comment by mr-c** [comment]
@luizirber I can haz merge?

**Comment by mr-c** [comment]
Thanks!

**Pull request merged**